### PR TITLE
add doxygen basics

### PR DESCRIPTION
**Pull request by byrnHDF**
Add basic files and structure for doxygen.
Generation and CI still under development

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -60,7 +60,7 @@ are tentative.
 
 HDF4 4.3.0 will clean up some long-standing issues in the library
 * The problematic XDR library will be replaced with a thin I/O layer that correctly handles long integer data on all platforms
-* We will drop support for FORTRAN 77 so the wrappers can be built like they are in HDF5
+* We will drop support for FORTRAN 77 so the wrappers can be built like they are in HDF4
```
Comment (bmribler) [inline README.md:63]: Is it supposed to be: "...like they are in HDF5"?

File: bin/release [hunk 2]
```diff
@@ -24,7 +24,7 @@ Usage: $0 -d <dir> [-h] [--private] [--revision [--branch BRANCHNAME]] <methods>
    -d DIR       The name of the directory where the release(s) should be
                 placed.
    --branch BRANCHNAME  This is to get the correct version of the branch name from the
-                        repository.  BRANCHNAME for v1.8 should be hdf5_1_8.
+                        repository.  BRANCHNAME for v1.8 should be hdf4_1_8.
```
Comment (bmribler) [inline bin/release:27]: This line is still not right for hdf4

File: release_notes/HISTORY.txt [hunk 3]
```diff
@@ -304,13 +304,13 @@ New features and changes
 
     - Removed all references to C++
 
-      HDF4 does not have a C++ wrapper and, like HDF5, we do not support
+      HDF4 does not have a C++ wrapper and, like HDF4, we do not support
```
Comment (bmribler) [inline release_notes/HISTORY.txt:307]: HDF5

File: release_notes/HISTORY.txt [hunk 4]
```diff
@@ -304,13 +304,13 @@ New features and changes
...
       building the library with a C++ compiler since the languages are
       diverging.
 
       Despite this, some parts of the build system checked for a C++ compiler,
       manipulated CXXFLAGS, etc. This was mainly in files that were brought
-      over from HDF5. This build system code has been removed.
+      over from HDF4. This build system code has been removed.
```
Comment (bmribler) [inline release_notes/HISTORY.txt:313]: HDF5

File: release_notes/HISTORY.txt [hunk 5]
```diff
@@ -1010,7 +1010,7 @@ New features and changes
 
       Added new configuration macros to use new HDF_BUILD_TYPE variable. This
       variable is set correctly for the type of generator being used for the build.
-      Changes follow from hdf5.
+      Changes follow from hdf4.
```
Comment (bmribler) [inline release_notes/HISTORY.txt:1013]: hdf5?

File: release_notes/HISTORY.txt [hunk 6]
```diff
@@ -1043,7 +1043,7 @@ Bugs fixed since HDF 4.2.13
       A number of improvements were made to the CMake commands. Most
       changes simplify usage or eliminate unused constructs. Also,
       some changes support better cross-platform support. Changes
-      follow from hdf5.
+      follow from hdf4.
```
Comment (bmribler) [inline release_notes/HISTORY.txt:1046]: hdf5?

File: release_notes/HISTORY.txt [hunk 7]
```diff
@@ -1057,7 +1057,7 @@ Bugs fixed since HDF 4.2.13
         GLOBAL PROPERTY -> GENERATOR_IS_MULTI_CONFIG
       Created two new HDF variable, HDF_BUILD_TYPE and HDF_CFG_BUILD_TYPE.
       Defaults for these variables is "Release". Changes
-      follow from hdf5.
+      follow from hdf4.
```
Comment (bmribler) [inline release_notes/HISTORY.txt:1060]: hdf5?

File: README.md (hunk 1 above)
Comment (byrnHDF) [reply README.md:63]: IDK, I just did a global S & R because of other locations.

Comment (byrnHDF) [reply README.md]: updated

File: bin/release (hunk 2 above)
Comment (byrnHDF) [reply bin/release]: never was, updated

File: release_notes/HISTORY.txt (hunk 7 above)
Comment (byrnHDF) [reply release_notes/HISTORY.txt]: I think I got them.

File: release_notes/HISTORY.txt [hunk 8]
```diff
@@ -251,7 +251,7 @@ New features and changes
       HDF4 uses pre-defined machine configurations instead of configure tests
       to include header files, set up typedefs, etc. In the future, the
       library will be adapted to use a flexible configure test scheme like
-      HDF5, but in the meantime we have simplified the machine configuration
+      HDF4, but in the meantime we have simplified the machine configuration
```
Comment (bmribler) [inline release_notes/HISTORY.txt:254]: HDF5

**Review by bmribler — APPROVED** [review]

**Review by lrknox — APPROVED** [review]

File: doxygen/dox/UsersGuide.dox [hunk 9]
```diff
@@ -0,0 +1,724 @@
+/** \page UG HDF User Guide
+
+<center>
+HDF Release 4.2.16
```
[the remark this answers is not in the document]
Comment (byrnHDF) [reply doxygen/dox/UsersGuide.dox:4]: These dox files are temporary, they will be updated properly.
However we should use a variable when we do update it.

**Pull request merged**